Scripts: Remove sourceURL from print_translations when not printing

sirreal opened the pull request:
Prevent adding `sourceURL` comment to script translations when not printed.

[In this comment](https://core.trac.wordpress.org/ticket/63887#comment:57), @ralucaStan reported an issue where a plugin uses `WP_Scripts::print_translations()` with `$display == false` to collect and concatenate multiple script tags into a single script tag. This reveals a possible regression and reveals that it's difficult to anticipate how the script tag contents may be used.

By not printing the `sourceURL` when `$display == false`, it's less likely to introduce problems for extenders in this case.

Trac ticket: https://core.trac.wordpress.org/ticket/63887
Follow-up to r60719.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

sirreal committed:
Prevent adding sourceURL to translation script if not printed

It's unclear how the script tag contents may be used by extenders. Avoid
printing the sourceURL if the contents are not immediately printed.

See
https://github.com/WordPress/wordpress-develop/blob/81e3dbeba63838a13b137c6308548e35a616f3b6/src/wp-includes/script-loader.php#L3015.

sirreal committed:
Include the sourceURL comment when printed by do_item

WP_Scripts uses print_translations() internally with $display == false.
In this case, the sourceURL comment is still desirable so include it.

westonruter commented on src/wp-includes/class-wp-scripts.php:
```diff
@@ -350,7 +350,16 @@ public function do_item( $handle, $group = false ) {
 		if ( $translations ) {
-			$translations = wp_get_inline_script_tag( $translations, array( 'id' => "{$handle}-js-translations" ) );
+			/**
+			 * The soucreURL comment is not included by @see WP_Scripts::print_translations()
```
```suggestion
			 * The sourceURL comment is not included by @see WP_Scripts::print_translations()
```

westonruter committed:
Fix typo

westonruter commented on src/wp-includes/class-wp-scripts.php:
```diff
@@ -350,7 +350,16 @@ public function do_item( $handle, $group = false ) {
+			/**
+			 * The sourceURL comment is not included by @see WP_Scripts::print_translations()
+			 * when `$display == false` to prevent issues where the script tag contents are used
+			 * by extenders for other purposes, for example concatenated with other script content.
```
Since this isn't the context for a docblock, I don't believe `/**` is the right comment style. Also, the bare `@see` isn't right. If used, I believe it would need to be `{@see WP_Scripts::print_translations()}` but @peterwilsoncc has informed me this shouldn't be used for functions.

```suggestion
			/*
			 * The sourceURL comment is not included by WP_Scripts::print_translations()
			 * when `$display === false` to prevent issues where the script tag contents are used
			 * by extenders for other purposes, for example concatenated with other script content.
```

sirreal replied:
You're right, the inline form of the see tag should be in braces `{@see …}`.

In my editor, I get proper highlighting and go-to definition if this is a PHPDoc comment (starts with `/**`) and I use the `@see` tag.

I've updated the comment. I can adjust further if necessary, but the best experience for me seems to be using `/** {@see …} */`.

peterwilsoncc replied:
@sirreal Yeah, it should only be a single asterisks as the parser can confuse the inline comment with a docblock, see https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments

sirreal replied:
Done, thanks for the clarification.

westonruter reviewed (approved):
Aside from the comment suggestion, this looks good to me. I haven't tested it, however.

sirreal committed:
Fix @see tag and comment text

peterwilsoncc commented:
It would be good to see a unit test on this to ensure `WP_Scripts::print_translations( ..., true )` has the source url and that `WP_Scripts::print_translations( ..., false )` does not.

sirreal committed:
Add test for no sourceURL when display is false

sirreal committed:
Remove phpdoc syntax from multiline comment

sirreal commented:
I cleaned up the comment and added a test to verify that `sourceURL=` is not present in the `print_translations()` return value when `$display === false`.

This should be ready to merge now.

sirreal commented:
This was merged in r61223.
